crypto/s2n_pq.c

```diff
@@ -17,12 +17,12 @@
 
 #include "crypto/s2n_openssl.h"
 
-bool s2n_libcrypto_supports_kyber()
+bool s2n_libcrypto_supports_evp_kem()
```

Comment:
I eliminated all usage (with a full code search) of Kyber where it was used to refer to something that used AWS-LC's EVP_KEM API to make it more generic long-term. There should no longer be any references to Kyber directly unless the code is explicitly talking about the Kyber-R3 algorithm. I decided to be more exhaustive exactly to avoid this confusion so I'm hoping there's no need for documentation of all this because there should be no more legacy nomenclature. EVP_KEM is the API, Kyber/ML-KEM are simply algorithms behind that API.